### PR TITLE
Add task relocation plan to task query result in gateway

### PR DESCRIPTION
**Pull request by tbak**

### CODE REVIEW

File: titus-server-runtime/src/main/java/com/netflix/titus/runtime/connector/relocation/client/GrpcAdapterRelocationServiceClient.java [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+
+/**
+ * {@link RelocationServiceClient} implementation that translates all invocation into underlying GRPC calls.
+ */
+@Singleton
+public class GrpcAdapterRelocationServiceClient implements RelocationServiceClient {
+
+    private static final Page ONE_ITEM_PAGE = Page.newBuilder().setPageSize(1).build();
+
+    private final ProtobufRelocationServiceClient protobufClient;
```
Comment (corindwyer) [inline titus-server-runtime/src/main/java/com/netflix/titus/runtime/connector/relocation/client/GrpcAdapterRelocationServiceClient.java]: grpcClient?

File: titus-server-runtime/src/main/java/com/netflix/titus/runtime/connector/relocation/client/GrpcAdapterRelocationServiceClient.java [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+                .setPage(ONE_ITEM_PAGE)
+                .putFilteringCriteria("taskIds", taskId)
+                .build()
+        ).flatMap(plans -> {
+            if (plans.getPlansList().isEmpty()) {
+                return Mono.just(Optional.empty());
+            }
+            // Sanity check
+            if (plans.getPlansList().size() > 1) {
+                return Mono.error(new IllegalStateException("Received multiple relocation plans for the same task id: " + taskId));
```
Comment (corindwyer) [inline titus-server-runtime/src/main/java/com/netflix/titus/runtime/connector/relocation/client/GrpcAdapterRelocationServiceClient.java]: relocations -> relocation

File: titus-server-gateway/src/main/java/com/netflix/titus/gateway/service/v3/JobManagerConfiguration.java [hunk 3]
```diff
@@ -37,4 +37,15 @@
...
+
+    /**
+     * The task relocation service has lower availability than TitusMaster. In case if it is down or very latent,
+     * applying the regular request timeout might result in large latency increase for all client requests.
+     * Instead we set more aggressive timeout for the task relocation service, at the cost of not including the
+     * relocation data in the client response.
+     *o
+     * @return fraction of the GRPC request timeout set for the task relocation service requests
+     */
+    @DefaultValue("0.3")
```
[the remark this answers is not in the document]
Comment (tbak) [reply titus-server-gateway/src/main/java/com/netflix/titus/gateway/service/v3/JobManagerConfiguration.java]: I have reduced this to 0.1 (== 1sec). This delay will happen only if there is availability issue with the task relocation component.

File: titus-server-runtime/src/main/java/com/netflix/titus/runtime/connector/relocation/client/GrpcAdapterRelocationServiceClient.java (hunk 1 above)
Comment (tbak) [reply titus-server-runtime/src/main/java/com/netflix/titus/runtime/connector/relocation/client/GrpcAdapterRelocationServiceClient.java]: This is a wrapper around GRPC client, but probably this name is less confusing than the current one.

File: titus-server-runtime/src/main/java/com/netflix/titus/runtime/connector/relocation/client/GrpcAdapterRelocationServiceClient.java (hunk 2 above)
Comment (tbak) [reply titus-server-runtime/src/main/java/com/netflix/titus/runtime/connector/relocation/client/GrpcAdapterRelocationServiceClient.java]: ?

**Pull request merged**